staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -65,27 +65,32 @@ type Validator struct {
 // Returns nil only if there no validator rules in the Structural schema. May return a validator containing
 // only errors.
 // Adding perCallLimit as input arg for testing purpose only. Callers should always use const PerCallLimit as input
-func NewValidator(s *schema.Structural, perCallLimit uint64) *Validator {
-	return validator(s, true, perCallLimit)
+func NewValidator(s *schema.Structural, isResourceRoot bool, perCallLimit uint64) *Validator {
+	return validator(s, true, model.SchemaDeclType(s, isResourceRoot), perCallLimit)
 }
 
-func validator(s *schema.Structural, isResourceRoot bool, perCallLimit uint64) *Validator {
-	compiledRules, err := Compile(s, isResourceRoot, perCallLimit)
+// validator creates a Validator for all x-kubernetes-validations at the level of the provided schema and lower and
+// returns the Validator if any x-kubernetes-validations exist in the schema, or nil if no x-kubernetes-validations
+// exist. declType is expected to be a CEL DeclType corresponding to the structural schema.
+func validator(s *schema.Structural, isResourceRoot bool, declType *model.DeclType, perCallLimit uint64) *Validator {
+	compiledRules, err := Compile(s, declType, perCallLimit)
 	var itemsValidator, additionalPropertiesValidator *Validator
 	var propertiesValidators map[string]Validator
 	if s.Items != nil {
-		itemsValidator = validator(s.Items, s.Items.XEmbeddedResource, perCallLimit)
+		itemsValidator = validator(s.Items, s.Items.XEmbeddedResource, declType.ElemType, perCallLimit)
 	}
 	if len(s.Properties) > 0 {
 		propertiesValidators = make(map[string]Validator, len(s.Properties))
 		for k, prop := range s.Properties {
-			if p := validator(&prop, prop.XEmbeddedResource, perCallLimit); p != nil {
-				propertiesValidators[k] = *p
+			if f, ok := declType.Fields[k]; ok {
```

Comment:
1. Oh boy. There are a couple problems with the escaping.  I've fixed and added tests to cover the different flows that were broken.
2. If there is a escaped or even an unescapable field, validation should still occur as normal.  Our only rule is that we don't allow CEL to access unescapable fields.  I've added tests covering these cases.  If we otherwise are unable to access a field that exists in the structural schema, that's an internal error.  Since we've moved all the code that could fail in this way to consumption, I believe those should be treated as an internal error (something unexpected has happened internally and we will need to fix whatever went wrong).
3. Desired behavior is:  CEL cannot access the field in rules, but we still traverse it while performing validation. I.e. a rule declared on a schema nested under an unescapable property name still should work as expected. This is fixed and working as expected now (and tested).